FIX: castToTemplate in case of empty template

SaarJunius opened the pull request:
In case the original data is an empty table (i.e. no rows), the castToTemplate function removed the first row of the adapted data.

This pr makes sure that castToTemplate works as expected for both filled and empty template data.

JasperSch commented:
I've checked with following test (which we should probably add under test-utils.R)

```
test_that("castToTemplate can cast from data.frame to tbl with empty template",{
			result <- castToTemplate(x = iris,  template = dplyr::as_tibble(iris[0,]))
			expect_equal(result, dplyr::as_tibble(iris))
		})
```

So the function doesn't remove a row anymore, but it also doesn't cast the table to the template.
Instead it returns the table as provided.

Hmm, not sure if we can somehow make an artificial row the trigger the normal behavior.

SaarJunius commented:
> I've checked with following test (which we should probably add under test-utils.R)
> 
> ```
> test_that("castToTemplate can cast from data.frame to tbl with empty template",{
> 			result <- castToTemplate(x = iris,  template = dplyr::as_tibble(iris[0,]))
> 			expect_equal(result, dplyr::as_tibble(iris))
> 		})
> ```
> 
> So the function doesn't remove a row anymore, but it also doesn't cast the table to the template. Instead it returns the table as provided.
> 
> Hmm, not sure if we can somehow make an artificial row the trigger the normal behavior.

That's a very useful test! I fixed the approach and I added this test (and similar ones for the other format conversions) to the test-utils.R file

JasperSch commented:
Lgtm, thank you!
